src/ReactiveUI/Platforms/ios/CommonReactiveSource.cs

```diff
@@ -410,9 +410,13 @@ private static IEnumerable<Update> GetUpdatesForEvent(PendingChange pendingChang
                         .Range(pendingChange.NewStartingIndex, pendingChange.NewItems == null ? 1 : pendingChange.NewItems.Count)
                         .Select(x => Update.CreateAdd(x));
                 case NotifyCollectionChangedAction.Remove:
+                    //Use OldStartingIndex for each "Update.Index" because the batch update processes and removes items sequentially
+                    //opposed to as one Range operation. 
+                    //For example if we are removing the items from indexes 1 to 5. 
+                    //When item at index 1 is removed item at index 2 is now at index 1 and so on down the line.
                     return Enumerable
                         .Range(pendingChange.OldStartingIndex, pendingChange.OldItems == null ? 1 : pendingChange.OldItems.Count)
-                        .Select(x => Update.CreateDelete(x));
+                        .Select(x => Update.CreateDelete(pendingChange.OldStartingIndex));
```

Comment:
Do we have any test covering this? Would be worth having one if possible -> if we do, that might also be a better place for the above comment to live.

Reply:
Possibly.. When I first wrote this there weren't any iOS specific tests but I see now there are. I built a repro project  https://github.com/PureWeen/RxUIReactiveTableViewSourceRepro.git  so would just need to integrate that concept into the tests.  I'll poke around next week see what I can do.

Reply:
If tests are a pain (I don't think we've got a concrete runner) don't worry too much. In future, I'm keen to get proper tests running on platforms, but that's a fairly long term aim.